apps/src/lib/node/ledger/ethereum_node/oracle.rs

```diff
@@ -32,6 +38,10 @@ pub struct Oracle {
     /// A channel to signal that the ledger should shut down
     /// because the Oracle has stopped
     abort: Option<Sender<()>>,
+    /// How long the oracle should wait between checking blocks
+    backoff: Duration,
+    /// If provided, the oracle will put here the latest block it has processed
+    latest_block_processed: Option<watch::Sender<Option<Uint256>>>,
```

Comment:
What if we made this `watch::Sender<Uint256>`? Then we don't need the other counter as this always retains the last height.

Reply:
will have a go at this and removing the other counter

Reply:
This is turning out to be pretty fiddly - while trying this approach, I noticed something that makes me think we should go back to having a channel where we publish every block height that is processed. So the other counter in `fn process()` is `next_block_to_process` which can start from `0` when the oracle is created - this current `latest_block_processed` fied starts from `None` then go `Some(0)`, `Some(1)`, `Some(2)`, etc. Replacing the field in this struct with something like:

```rust
next_block_to_process: watch::Sender<Uint256>
```

works and gets rid of the `Option`s, and we can just use this one field to keep track, but then in `test_all_blocks_checked` when we set the height straightaway to be some height more than `MIN_CONFIRMATIONS`, the oracle can start processing blocks immediately and the `next_block_to_process` field can be quickly updated to some block more than 0. This is also an issue with the current `Option<watch::Sender<Option<Uint256>>>` approach as well - if I add a sleep in that test, the oracle races ahead.

Reply:
This is what I tried - https://github.com/anoma/namada/commit/302a0f2322dd46e4f1de1314c101d892adacd972

Reply:
I'm not sure I understand the issue as you've described it

Reply:
Ah, I see! The latest block processessed shouldn't start at 0. It should query geth to get the latest block height and set `next_block_to_process` to that. And that should be the last time it queries the block height from geth. So then it just increments the block height one by one.

Reply:
Yeah something like this needs to be done (have opened https://github.com/anoma/namada/issues/589). The issue for the tests is that we want to see every block height that was checked and in order, not just the `latest_block_processed` or `next_block_to_process`, so that we can assert the oracle is checking all blocks in sequence. I'm not sure of the best way to do this but a channel which the oracle sends block heights it has checked works, or we could provide a callback to the oracle for it to notify us when it has processed a block (that we can then use in our tests).